src/plugins/intel_gpu/src/graph/graph_optimizer/mark_shape_of_subgraphs.cpp

```diff
@@ -86,6 +87,13 @@ bool mark_shape_of_subgraphs::can_mark_node(const program_node& node) {
         return false;
     }
 
+    // skip mark_node for broadcast node if dependency nodes are data and shape_of
```

Comment:
Could you please add related test?

Reply:
Done. Testcases added for convolution w/o broadcast in commit https://github.com/openvinotoolkit/openvino/pull/27786/commits/f25e82e66444b9107589c32eccb7196795a53438.